src/riak_cs_simple_quota.erl

```diff
@@ -0,0 +1,218 @@
+    catch ets:insert_new(Table, UserState),
+    UserState.
+
+test_usage_state(#user_state{usage=Usage} = _,
+                 #state{quota=Quota}) when Quota > 0 ->
+    case Usage < Quota of
+        true ->
+            ok;
+        false ->
+            {error, {disk_quota, Usage, Quota}}
+    end;
+test_usage_state(_, _) ->
+    ok.
+
+update(User,
+       #wm_log_data{method = 'PUT',
+                    headers = Headers} = _LogData,
+       State) ->
+    #state{table=Table} = State,
+    RequestLength = mochiweb_headers:get_value("content-length", Headers),
+    Bytes = case catch list_to_integer(RequestLength) of
+                Len when is_integer(Len) -> Len;
+                _ -> 0
+            end,
+    try
+        ets:update_counter(Table, User, {3, Bytes})
+    catch Type:Error ->
+            %% TODO: show out stacktrace heah
+            lager:warning("something wrong? ~p", [Error]),
+            {error, {Type, Error}}
+    end;
+update(_, _, _) -> ok.
+
+
+handle_info(refresh, #state{table=Table, duration_sec=Sec} = State) ->
+    ets:delete_all_objects(Table),
+    _ = maybe_schedule_refresh(Sec),
+    {noreply, State};
+handle_info(_, State) ->
+    {noreply, State}.
+
+update_state(Data, #state{table=Table} = _State) ->
+    ets:delete_all_objects(Table),
+    {DurationSec, UserQuota} = Data,
+    _ = maybe_schedule_refresh(DurationSec),
+    {ok, #state{table=Table,
+                duration_sec = DurationSec,
+                quota = UserQuota}}.
+
+terminate(#state{table=Table} = _State) ->
+    true = ets:delete(Table),
+    ok.
+
+maybe_schedule_refresh(DurationSec) when DurationSec > 0 ->
+    Self = self(),
+    erlang:send_after(DurationSec*1000, Self, refresh);
+maybe_schedule_refresh(_) ->
+    ignore.
+
+get_latest_usage(User) ->
+    {ok, Pid} = riak_cs_riak_client:checkout(),
+    Now = calendar:now_to_datetime(os:timestamp()),
+    NowASec = calendar:datetime_to_gregorian_seconds(Now),
+    try
+        get_latest_usage(Pid, User, NowASec, 0, 10)
+    after
+            riak_cs_riak_client:checkin(Pid)
+    end.
+
+get_latest_usage(_Pid, _User, _, N, N) -> {error, notfound};
+get_latest_usage(Pid, User, EndSec, N, Max) ->
+    End = calendar:gregorian_seconds_to_datetime(EndSec),
+    ADayAgo = calendar:gregorian_seconds_to_datetime(EndSec - 86400),
+    case riak_cs_storage:get_usage(Pid, User, ADayAgo, End) of
+        {[], _} ->
+            get_latest_usage(Pid, User, EndSec - 86400, N+1, Max);
+        {Res, _} ->
+            lager:info("Res: ~p", [Res]),
+            {ok, Res}
+    end.
+
+sum_all_buckets(UsageList) ->
+    lists:foldl(fun({<<"StartTime">>, _}, Sum) -> Sum;
+                   ({<<"EndTime">>, _}, Sum) -> Sum;
+                   ({_BucketName, {struct, Data}}, Sum) ->
+                        case proplists:get_value(<<"Bytes">>, Data) of
+                            I when is_integer(I) -> Sum + I;
+                            _ -> Sum
+                        end
```

Comment:
I don't care completeness because this module is an example implementation.
If someone use it or implement his/her own one, please add clause `{BucketName, ErrorMsgBin}`.